homeassistant/components/usb/__init__.py

```diff
@@ -41,6 +43,24 @@ class UsbServiceInfo(TypedDict):
     manufacturer: str | None
     description: str | None
 
+    # Used to prevent log flooding. To be removed in 2022.6
+    _warning_logged: bool = False
+
+    def __getitem__(self, name: str) -> Any:
+        """
+        Allow property access by name for compatibility reason.
+
+        Deprecated, and will be removed in version 2022.6.
+        """
+        if not self._warning_logged:
+            report(
+                f"accessed discovery_info['{name}'] instead of discovery_info.{name}; this will fail in version 2022.6",
+                exclude_integrations={"usb"},
+                error_if_core=False,
+            )
```

Comment:
I think it is ok to use `WARNING` not `DEBUG` for `usb` as there are only three components:
- `modem_callerid`
- `zha`
- `zwave_js`

Reply:
That works

Reply:
@epenet 

Actually since all 3 of them will still be triggering the warning until they are updated, we should leave it at debug until they are actually changed in the codebase

Reply:
OK, I've updated it.